File: dev-docs/snippets/rule-sets/get.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Retrieve a single rule set by its ID.
export async function get(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const ruleSet = await client.ruleSets.get(ruleSetId);
  // #endregion example
  return ruleSet;
}

File: dev-docs/snippets/rule-sets/get.swift
import JsBaoClient

// Retrieve a single rule set by its ID. Swift returns a typed `RuleSetInfo`.
func get(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let ruleSet = try await client.ruleSets.get(ruleSetId: ruleSetId)
  // #endregion example
  _ = ruleSet
}
